src/Symfony/Component/Routing/Tests/Fixtures/dumper/url_matcher4.php

```diff
@@ -21,82 +21,54 @@ public function match($rawPathinfo)
-                $allow[] = 'GET';
-                goto not_head_and_get;
-            }
-
-            return array('_route' => 'head_and_get');
-        }
-        not_head_and_get:
-
-        // get_and_head
-        if ('/get_and_head' === $pathinfo) {
-            if ('GET' !== $canonicalMethod) {
-                $allow[] = 'GET';
-                goto not_get_and_head;
-            }
-
-            return array('_route' => 'get_and_head');
-        }
-        not_get_and_head:
-
-        // post_and_head
-        if ('/post_and_get' === $pathinfo) {
-            if (!in_array($requestMethod, array('POST', 'HEAD'))) {
-                $allow = array_merge($allow, array('POST', 'HEAD'));
-                goto not_post_and_head;
-            }
-
-            return array('_route' => 'post_and_head');
-        }
-        not_post_and_head:
-
-        if (0 === strpos($pathinfo, '/put_and_post')) {
-            // put_and_post
-            if ('/put_and_post' === $pathinfo) {
-                if (!in_array($requestMethod, array('PUT', 'POST'))) {
-                    $allow = array_merge($allow, array('PUT', 'POST'));
+        switch ($pathinfo) {
+            case '/put_and_post':
+                // put_and_post
+                $ret = array('_route' => 'put_and_post');
+                if (!isset(($a = array('PUT' => 0, 'POST' => 1))[$requestMethod])) {
+                    $allow += $a;
                     goto not_put_and_post;
                 }
 
-                return array('_route' => 'put_and_post');
-            }
-            not_put_and_post:
-
-            // put_and_get_and_head
-            if ('/put_and_post' === $pathinfo) {
-                if (!in_array($canonicalMethod, array('PUT', 'GET'))) {
-                    $allow = array_merge($allow, array('PUT', 'GET'));
+                return $ret;
+                not_put_and_post:
+                // put_and_get_and_head
+                $ret = array('_route' => 'put_and_get_and_head');
+                if (!isset(($a = array('PUT' => 0, 'GET' => 1, 'HEAD' => 2))[$canonicalMethod])) {
+                    $allow += $a;
                     goto not_put_and_get_and_head;
                 }
 
-                return array('_route' => 'put_and_get_and_head');
-            }
-            not_put_and_get_and_head:
+                return $ret;
+                not_put_and_get_and_head:
+                break;
+            default:
+                $routes = array(
+                    '/just_head' => array(array('_route' => 'just_head'), null, array('HEAD' => 0), null),
+                    '/head_and_get' => array(array('_route' => 'head_and_get'), null, array('HEAD' => 0, 'GET' => 1), null),
+                    '/get_and_head' => array(array('_route' => 'get_and_head'), null, array('GET' => 0, 'HEAD' => 1), null),
+                    '/post_and_get' => array(array('_route' => 'post_and_head'), null, array('POST' => 0, 'HEAD' => 1), null),
```

Comment:
the path uses post_and_GET while the route is named post_and_HEAD which is inconsistent

Reply:
fixed (on 3.4 also)